storage/rocksdb/rdb_perf_context.h

```diff
@@ -17,13 +17,109 @@
 #pragma once
 
 /* C++ standard header files */
+#include <algorithm>
 #include <atomic>
 #include <cstdint>
 #include <string>
 
 /* MySQL header files */
 #include "./handler.h"
 #include <my_global.h>
+#include "./atomic_stat.h"
+
+/* my_io_perf_* functionality snatched from upstream FacebookSQL server core */
+
+/* Struct used for IO performance counters within a single thread */
+struct my_io_perf_struct {
+  ulonglong bytes;
+  ulonglong requests;
+  ulonglong svc_time; /*!< time to do read or write operation */
+  ulonglong svc_time_max;
+  ulonglong wait_time; /*!< total time in the request array */
+  ulonglong wait_time_max;
+ulonglong slow_ios; /*!< requests that take too long */
+};
+typedef struct my_io_perf_struct my_io_perf_t;
+
+/* Struct used for IO performance counters, shared among multiple threads */
+struct my_io_perf_atomic_struct {
+  atomic_stat<ulonglong> bytes;
+  atomic_stat<ulonglong> requests;
+  atomic_stat<ulonglong> svc_time; /*!< time to do read or write operation */
+  atomic_stat<ulonglong> svc_time_max;
+  atomic_stat<ulonglong> wait_time; /*!< total time in the request array */
+  atomic_stat<ulonglong> wait_time_max;
+  atomic_stat<ulonglong> slow_ios; /*!< requests that take too long */
+};
+typedef struct my_io_perf_atomic_struct my_io_perf_atomic_t;
+
+/* Initialize an my_io_perf_t struct. */
+static inline void my_io_perf_init(my_io_perf_t* perf) {
```

Comment:
OK, I suggest checking with the upstream if they'd accept the patch and going for it